### PR TITLE
fix(discover): fix Go to Summary sometimes redirecting to performance landing

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
The "Go to Summary" cell action on the transaction column only works if `project` is a selected column. Otherwise, it gets redirected to the performance landing page.

Fixes this issue by using `project.name` which is provided by eventsV2 instead of expecting the user to have selected the `project` column

### CODE REVIEW

File: static/app/views/eventsV2/table/tableView.tsx [hunk 1]
```diff
@@ -373,8 +373,9 @@ class TableView extends React.Component<TableViewProps> {
 
       switch (action) {
         case Actions.TRANSACTION: {
-          const maybeProject = projects.find(project => project.slug === dataRow.project);
-
+          const maybeProject = projects.find(
```
Comment (shruthilayaj) [inline static/app/views/eventsV2/table/tableView.tsx:376]: Seems like this fixes the case where the project column isn't selected but if project is selected as a column, the "Go to summary" now goes to the performance landing page

**Review by shruthilayaj — COMMENTED** [review]
Should we add a test to ensure broweserHistory is called with the expected project id?

File: static/app/views/eventsV2/table/tableView.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/views/eventsV2/table/tableView.tsx:376]: Hey Shruthi. Hmm, I did test with `project` selected as a column and `eventsV2` does still return the `project.name` value in the response ie "Go to summary" still goes to the correct Transaction Summary page.

Here's the columns I selected to test with:
image

let me know if I missed a test case here!

Comment (edwardgou-sentry) [reply static/app/views/eventsV2/table/tableView.tsx:376]: Updated tests and updated to support `project` or `project.name`!

**Review by shruthilayaj — APPROVED** [review]

**Pull request merged**